library/std/src/io/mod.rs

```diff
@@ -815,6 +817,30 @@ pub trait Read {
     /// file.)
     ///
     /// [`std::fs::read`]: crate::fs::read
+    ///
+    /// ## Implementing `read_to_end`
+    ///
+    /// When implementing the `io::Read` trait, it is recommended to allocate
+    /// memory using [`Vec::try_reserve`]. However, this behavior is not guaranteed
+    /// by all implementations, and `read_to_end` may not handle out-of-memory
+    /// situations gracefully.
```

Comment:
Should we mention that the default impl provides this behavior?

Reply:
I've documented that the default implementation uses `try_reserve`. I don't see any problems with promising it, since it's easy from implementation perspective, and `try_reserve` gets the blame for all of the OOM detection complexity.